src/hc32f460/serial.c

```diff
@@ -0,0 +1,189 @@
+// HC32F460 serial
+//
+// Copyright (C) 2022  Steven Gotthardt <gotthardt@gmail.com>
+//
+// This file may be distributed under the terms of the GNU GPLv3 license.
+
+#include "autoconf.h" // CONFIG_SERIAL_BAUD
+
+#include "command.h"    // DECL_CONSTANT_STR
+#include "internal.h"
+#include "sched.h"      // DECL_INIT
+#include "generic/serial_irq.h"
+#include "generic/armcm_boot.h"
+
+#include "hc32f460_usart.h"
+#include "hc32f460_gpio.h"
+#include "hc32f460_pwc.h"
+
+
+#define USART_BAUDRATE  (CONFIG_SERIAL_BAUD)
+
+
+// Aquila 1.0.3 pins for TX, RX to CH304 on PA15 and PA09
+//    for the LCD connector: TX, RX on PC00 and PC01
+#if CONFIG_HC32F460_SERIAL_PA15_PA9
+  DECL_CONSTANT_STR("RESERVE_PINS_serial", "PA15,PA9");
+  #define USART_RX_PORT   (PortA)
+  #define USART_RX_PIN    (Pin15)
+  #define USART_TX_PORT   (PortA)
+  #define USART_TX_PIN    (Pin09)
+
+#elif CONFIG_HC32F460_SERIAL_PC0_PC1
+  DECL_CONSTANT_STR("RESERVE_PINS_serial", "PC0,PC1");
+  #define USART_RX_PORT   (PortC)
+  #define USART_RX_PIN    (Pin00)
+  #define USART_TX_PORT   (PortC)
+  #define USART_TX_PIN    (Pin01)
+
+#elif CONFIG_HC32F460_SERIAL_PA3_PA2
+  DECL_CONSTANT_STR("RESERVE_PINS_serial", "PA3,PA2");
+  #define USART_RX_PORT   (PortA)
+  #define USART_RX_PIN    (Pin03)
+  #define USART_TX_PORT   (PortA)
+  #define USART_TX_PIN    (Pin02)
+
+#elif CONFIG_HC32F460_SERIAL_PA7_PA8
+  DECL_CONSTANT_STR("RESERVE_PINS_serial", "PA7,PA8");
+  #define USART_RX_PORT   (PortA)
+  #define USART_RX_PIN    (Pin07)
+  #define USART_TX_PORT   (PortA)
+  #define USART_TX_PIN    (Pin08)
+#endif
+
+// use USART 1 for serial connection
+#define USARTx          M4_USART1
+#define USART_ENABLE    PWC_FCG1_PERIPH_USART1 | PWC_FCG1_PERIPH_USART2 | \
+                        PWC_FCG1_PERIPH_USART3 | PWC_FCG1_PERIPH_USART4
```

Comment:
Almost certainly this define should be wrapped in parenthesis.

Reply:
wrapped